Samples/Hast.Samples.SampleAssembly/ImageSharpSample.cs

```diff
@@ -0,0 +1,127 @@
+    /// </summary>
+    public class ImageSharpSample
+    {
+        private const int Resize_ImageWidthIndex = 0;
+        private const int Resize_ImageHeightIndex = 1;
+        private const int Resize_DestinationImageWidthIndex = 2;
+        private const int Resize_DestinationImageHeightIndex = 3;
+        private const int Resize_FrameCount = 4;
+        private const int Resize_ImageStartIndex = 5;
+
+        [Replaceable(nameof(ImageSharpSample) + "." + nameof(MaxDegreeOfParallelism))]
+        private static readonly int MaxDegreeOfParallelism = 25;
+
+        public virtual void ApplyTransform(SimpleMemory memory)
+        {
+            var width = (ushort)memory.ReadUInt32(Resize_ImageWidthIndex);
+            var height = (ushort)memory.ReadUInt32(Resize_ImageHeightIndex);
+            var destWidth = (ushort)memory.ReadUInt32(Resize_DestinationImageWidthIndex);
+            var destHeight = (ushort)memory.ReadUInt32(Resize_DestinationImageHeightIndex);
+            var frameCount = (ushort)memory.ReadUInt32(Resize_FrameCount);
+            var destinationStartIndex = width * height + 5;
+
+            var widthFactor = width / destWidth;
+            var heightFactor = height / destHeight;
+
+            var tasks = new Task<PixelProcessingTaskOutput>[MaxDegreeOfParallelism];
+
+            var stepCount = destWidth / MaxDegreeOfParallelism;
+            var wastedSteps = destWidth % MaxDegreeOfParallelism;
+
+            if (wastedSteps != 0)
+            {
+                stepCount += 1;
+            }
+
+            for (int f = 0; f < frameCount; f++)
+            {
+                for (int y = 0; y < destHeight; y++)
+                {
+                    for (int x = 0; x < stepCount; x++)
+                    {
+                        for (int t = 0; t < MaxDegreeOfParallelism; t++)
+                        {
+                            if (x * widthFactor * MaxDegreeOfParallelism + t * widthFactor >= width)
+                            {
+                                break;
+                            }
+
+                            var pixelBytes = memory.Read4Bytes(
+                                (1 + f) * (y * heightFactor * destWidth * heightFactor
+                                    + x * widthFactor * MaxDegreeOfParallelism
+                                    + t * widthFactor) + Resize_ImageStartIndex);
+
+                            tasks[t] = Task.Factory.StartNew(inputObject =>
+                            {
+                                var input = (PixelProcessingTaskInput)inputObject;
+
+                                return new PixelProcessingTaskOutput
+                                {
+                                    R = input.PixelBytes[0],
+                                    G = input.PixelBytes[1],
+                                    B = input.PixelBytes[2],
+                                    A = input.PixelBytes[3]
+                                };

```

Comment:
Itt valójában nem történik semmi. A Hastlayerrrel a lényeg, hogy párhuzamosítva futtassunk le egy compute-bound algoritmust. A resize logikának kellene tehát úgy lefutnia. Ezt kellene tehát lefuttatni: https://github.com/Lombiq/Hastlayer-ImageSharp-PracticeDemo/blob/7910bb2c33a8b71da64246e4b89ba914c1bd9d95/Hastlayer-ImageSharp-PracticeDemo/Resize/HastResizeProcessor%7BTPixel%7D.cs#L202

Itt az egész megközelítésnél elvi hibát látok: ennek az egésznek úgy kell működnie (ahogy erről beszéltünk is régebben), hogy meghívód az ImageSharp akármilyét, ami a resize-t elkezdi, és azt úgy configolod fel, hogy valahol belül a pipeline részeként lefuttassa a Hastlayerrel gyorsított kódot. Utóbbi, ami a `GenerateProxy()`-ból kijön.

Itt a többi sample-hez képest annyiban bonyolultabb, hogy nem te hívod meg ezt egyből, hanem az ImageSharpot hívod meg, ami aztán meghívja a hastlayeres kódot. De a lényegen ennek nem kell változtatnia.